src/App.js

```diff
@@ -1,25 +1,37 @@
-import React, {useState} from 'react'; 
+import React, {useState,useEffect} from 'react'; 
 import TodoList from './TodoList'; //remember the ./ because it's in a different file
 import AddTodoForm from './AddTodoForm';
 
-
+const useSemiPersistentState=()=>{ 
+  const [todoList,setTodoList]=useState(JSON.parse(localStorage.getItem('savedTodoList')))
+    // the todoList is the savedTodoList 
+    // parsing to make string an object
+    useEffect(()=>{ 
+      localStorage.setItem('savedTodoList',JSON.stringify(todoList))
+    },[todoList])
+    // made todoList a string
+    // savedTodoList is key and todoList is value
+    // each time todoList is change we are saving to savedTodoList in localStorage
+      return( 
```

Comment:
nit: you generally only need parentheses when you're doing a multi line return. This should be fine on one line so you can do 

`return  [todoList,setTodoList]`